compiler/rustc_ast/src/mut_visit.rs

```diff
@@ -954,7 +954,12 @@ fn walk_coroutine_kind<T: MutVisitor>(vis: &mut T, coroutine_kind: &mut Coroutin
 
 fn walk_fn<T: MutVisitor>(vis: &mut T, kind: FnKind<'_>) {
     match kind {
-        FnKind::Fn(_ctxt, _ident, FnSig { header, decl, span }, _visibility, generics, body) => {
+        FnKind::Fn(
+            _ctxt,
+            _ident,
+            _vis,
+            Fn { generics, body, sig: FnSig { header, decl, span }, .. },
```

Comment:
exhaustively match here so future additions won't be missed

Reply:
My bad... I was in the mode of simplifying code, and I forgot that for visitors we keep match explicit so the visitor is always up to date.